homeassistant/components/energy/data.py

```diff
@@ -58,13 +58,22 @@ class FlowToGridSourceType(TypedDict):
     number_energy_price: float | None  # Price for energy ($/kWh)
 
 
+class GridPowerSourceType(TypedDict):
+    """Dictionary holding the source of grid power consumption."""
+
+    # statistic_id of an power meter (kW)
+    # negative values indicate grid return
+    stat_power: str
+
+
 class GridSourceType(TypedDict):
     """Dictionary holding the source of grid energy consumption."""
 
     type: Literal["grid"]
 
     flow_from: list[FlowFromGridSourceType]
     flow_to: list[FlowToGridSourceType]
+    power: list[GridPowerSourceType] | None
```

Comment:
Is the typing correct? Will this key always be present but `None` sometimes? I don't see that in any of the websocket tests.